src/include/serial_inline.h

```diff
@@ -229,6 +229,9 @@ __wt_update_serial(WT_SESSION_IMPL *session, WT_CURSOR_BTREE *cbt, WT_PAGE *page
     upd = *updp;
     *updp = NULL;
     prev_upd_ts = WT_TS_NONE;
+#ifdef HAVE_DIAGNOSTIC
+    prev_upd_ts = upd->prev_durable_ts;
+#endif
```

Comment:
I see what is wrong here. May be we can remove line 231 and directly initialize it to upd->prev_durable_ts. There is no need to wrap it in HAVE_DIAGNOSTIC macro.

Reply:
That's the first thing I tried. I learned that `__wt_update::prev_durable_ts` is defined in the `HAVE_DIAGNOSTIC` ifdef
https://github.com/mongodb/mongo/blob/e2395982273ecf05cb5dd9373ed106e025fb4916/src/third_party/wiredtiger/src/include/btmem.h#L1109

Reply:
That's correct.